### PR TITLE
Home automation app unsolved

### PR DESCRIPTION
**Pull request by jramcast**
Code for https://github.com/RedHatTraining/DO400/issues/70.

The app is just a skeleton. Students must write all code

### CODE REVIEW

**Review by gjbianco — APPROVED** [review]

File: home-automation/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+*
+!target/*-runner
+!target/*-runner.jar
+!target/lib/*
+!target/quarkus-app/*
```
Comment (psolarvi) [inline home-automation/.dockerignore:5]:
```suggestion
!target/quarkus-app/*

```

File: home-automation/.gitignore [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# patch
+*.orig
+*.rej
+
+# Maven
+target/
+pom.xml.tag
+pom.xml.releaseBackup
+pom.xml.versionsBackup
+release.properties
```
Comment (psolarvi) [inline home-automation/.gitignore:36]:
```suggestion
release.properties

```

File: home-automation/src/main/docker/Dockerfile.fast-jar [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# We make four distinct layers so if there are application changes the library layers can be re-used
+COPY --chown=1001 target/quarkus-app/lib/ /deployments/lib/
+COPY --chown=1001 target/quarkus-app/*.jar /deployments/
+COPY --chown=1001 target/quarkus-app/app/ /deployments/app/
+COPY --chown=1001 target/quarkus-app/quarkus/ /deployments/quarkus/
+
+EXPOSE 8080
+USER 1001
+
+ENTRYPOINT [ "/deployments/run-java.sh" ]
```
Comment (psolarvi) [inline home-automation/src/main/docker/Dockerfile.fast-jar:57]:
```suggestion
ENTRYPOINT [ "/deployments/run-java.sh" ]

```

File: home-automation/src/main/docker/Dockerfile.jvm [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+# Configure the JAVA_OPTIONS, you can add -XshowSettings:vm to also display the heap size.
+ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
+
+COPY target/lib/* /deployments/lib/
+COPY target/*-runner.jar /deployments/app.jar
+
+EXPOSE 8080
+USER 1001
+
+ENTRYPOINT [ "/deployments/run-java.sh" ]
```
Comment (psolarvi) [inline home-automation/src/main/docker/Dockerfile.jvm:54]:
```suggestion
ENTRYPOINT [ "/deployments/run-java.sh" ]

```

File: home-automation/src/main/docker/Dockerfile.native [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+WORKDIR /work/
+RUN chown 1001 /work \
+    && chmod "g+rwX" /work \
+    && chown 1001:root /work
+COPY --chown=1001:root target/*-runner /work/application
+
+EXPOSE 8080
+USER 1001
+
+CMD ["./application", "-Dquarkus.http.host=0.0.0.0"]
```
Comment (psolarvi) [inline home-automation/src/main/docker/Dockerfile.native:27]:
```suggestion
CMD ["./application", "-Dquarkus.http.host=0.0.0.0"]

```

File: home-automation/src/main/resources/META-INF/resources/index.html [hunk 6]
```diff
@@ -0,0 +1,155 @@
+<!DOCTYPE html>
```
Comment (psolarvi) [inline home-automation/src/main/resources/META-INF/resources/index.html:1]: This file can be removed

File: home-automation/src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,2 @@
+# Configuration file
+# key = value
```
Comment (psolarvi) [inline home-automation/src/main/resources/application.properties:2]:
```suggestion
# key = value

```

**Review by psolarvi — APPROVED** [review]

**Pull request merged**